lib/spack/env/cc

```diff
@@ -142,6 +151,12 @@ if [[ -z $mode ]]; then
     done
 fi
 
+# Set up rpath variable according to language.
+eval rpath=\$SPACK_${comp}_RPATH_ARG
+if [[ $rpath != *%s* ]]; then
+    die "SPACK_${comp}_RPATH_ARG must contain '%s' for parameter substitution."
```

Comment:
The intent is to support forms where the argument is not final.  I am not sure they exist though... thoughts?